symbolic-base/src/ZkFold/Base/Protocol/IVC/VerifierCircuit.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TypeOperators       #-}
+
+module ZkFold.Base.Protocol.IVC.VerifierCircuit where
+
+-- import           GHC.Generics                        (Par1 (..), U1 (..), type (:.:) (..), (:*:) (..))
+-- import           Prelude                             hiding (Num (..), drop, head, replicate, take, zipWith)
+
+-- import           ZkFold.Base.Algebra.Basic.Number    (KnownNat, type (+), type (-))
+-- import           ZkFold.Base.Data.ByteString         (Binary)
+-- import           ZkFold.Base.Data.Vector             (Vector)
+-- import           ZkFold.Base.Protocol.IVC.Commit     (HomomorphicCommit)
+-- import           ZkFold.Base.Protocol.IVC.CommitOpen (CommitOpen (..))
+-- import           ZkFold.Base.Protocol.IVC.FiatShamir (FiatShamir (..))
+-- import           ZkFold.Base.Protocol.IVC.Internal   (IVCResult, ivcVerify, RecursiveI, RecursiveP)
+-- import           ZkFold.Base.Protocol.IVC.Oracle     (HashAlgorithm)
+-- import           ZkFold.Base.Protocol.IVC.Predicate  (Predicate (..), predicate)
+-- import           ZkFold.Symbolic.Class
+-- import           ZkFold.Symbolic.Compiler
+-- import           ZkFold.Symbolic.Data.Class          (SymbolicData (..))
+-- import           ZkFold.Symbolic.Data.FieldElement   (FieldElement (..))
+-- import           ZkFold.Symbolic.Data.Input          (SymbolicInput)
+-- import ZkFold.Base.Protocol.IVC.SpecialSound (SpecialSoundProtocol(..))
+
+-- -- | Takes a function `f` and returns a circuit `C` with input `y` and witness `w`.
+-- -- The circuit is such that `C(y, w) = 0` implies that `y = x(n)` for some positive `n` where
+-- -- `x(k+1) = f(x(k), u(k))` for all `k` and some `u`.
+-- ivcVerifierCircuit :: forall f i p m c d k a payload input output nx nu ctx algo .
```

Comment:
So this would be re-added in some other way, right?

Reply:
Yes